pkg/cmd/pulumi/install/install.go

```diff
@@ -216,3 +237,34 @@ func shouldInstallPolicyPackDependencies() (bool, error) {
 	}
 	return false, nil
 }
+
+func shouldInstallPluginDependencies() (bool, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return false, fmt.Errorf("getting the working directory: %w", err)
+	}
+	pluginPath, err := workspace.DetectPluginPathFrom(cwd)
+	if err != nil {
+		return false, fmt.Errorf("detecting policy pack path: %w", err)
+	}
+	if pluginPath != "" {
+		// There's a PulumiPlugin.yaml in cwd or a parent folder. The plugin might be nested
+		// within a project, or vice-vera, so we need to check if there's a Pulumi.yaml in a parent
+		// folder.
+		projectPath, err := workspace.DetectProjectPathFrom(cwd)
+		if err != nil {
+			if errors.Is(err, workspace.ErrProjectNotFound) {
+				// No project found, we should install the dependencies for the policy pack.
+				return true, nil
+			}
+			return false, fmt.Errorf("detecting project path: %w", err)
+		}
+		// We have both a project and a policy pack. If the project path is a parent of the policy
+		// pack path, we should install dependencies for the policy pack, otherwise we should
```

Comment:
```suggestion
		// We have both a project and a plugin. If the project path is a parent of the plugin
		// path, we should install dependencies for the plugin, otherwise we should
```